llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -706,6 +710,12 @@ class VPRecipeWithIRFlags : public VPSingleDefRecipe {
       : VPSingleDefRecipe(SC, Operands, DL), OpType(OperationType::DisjointOp),
         DisjointFlags(DisjointFlags) {}
 
+  template <typename IterT>
+  VPRecipeWithIRFlags(const unsigned char SC, IterT Operands,
+                      NonNegFlagsTy NonNegFlags, DebugLoc DL = {})
```

Comment:
nit: 
I'm wondering if it makes sense to make `NonNegFlags` a `std::optional<NonNegFlagsTy>`, and initialise it with `{}` by default (here and in the other classes), so that you don't need to do:
```
if (Something->isZExt())
  ... = new VPWidenCastRecipe(..., Something->isNonNeg());
else
  ... = new VPWidenCastRecipe(...);
```
and instead just do:
```
std::optional<NonNegFlagsTy> NonNegFlags = Something->isZExt() ? Something->isNonNeg() : {};
... = new VPWidenCastRecipe(..., NonNegFlags);
```
What do you think?

Reply:
You're right, ignore my suggestion, this isn't as easy as I thought it was.